src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -188,17 +188,18 @@ public Task StopAsync()
 
         public virtual async Task WriteAsync(ServiceMessage serviceMessage)
         {
-            var errorMessage = _errorMessage;
-            if (!string.IsNullOrEmpty(errorMessage))
+            if (!await SafeWriteAsync(serviceMessage))
             {
-                // Can be out of the lock, it is ok to send out messages even when _error received.
-                throw new ServiceConnectionNotActiveException(errorMessage);
+                throw new ServiceConnectionNotActiveException(_errorMessage);
             }
+        }
 
-            if (Status != ServiceConnectionStatus.Connected)
+        protected async Task<bool> SafeWriteAsync(ServiceMessage serviceMessage)
+        {
+            var errorMessage = _errorMessage;
+            if (!string.IsNullOrEmpty(errorMessage) || Status != ServiceConnectionStatus.Connected)
```

Comment:
>errorMessage [](start = 38, length = 12)

could here errorMessage different from the one throws?

Reply:
maybe, the latest one matters, sounds make sense too